From 12569efa62f81a89e98268787a12eafe3f3bacfa Mon Sep 17 00:00:00 2001
From: Laszlo Ersek <lersek@redhat.com>
Date: Thu, 13 Nov 2014 15:18:42 +0100
Subject: [PATCH 7/7] ArmPlatformPkg/Bds: FindCandidateOnHandle(): log full
 device path

Since we scan several handles / devices now, log the full device path when
we find the candidate.

Contributed-under: TianoCore Contribution Agreement 1.0
Signed-off-by: Laszlo Ersek <lersek@redhat.com>
---
 ArmPlatformPkg/Bds/Bds.c   | 10 +++++++++-
 ArmPlatformPkg/Bds/Bds.inf |  1 +
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/ArmPlatformPkg/Bds/Bds.c b/ArmPlatformPkg/Bds/Bds.c
index f255a25..0bef8f0 100644
--- a/ArmPlatformPkg/Bds/Bds.c
+++ b/ArmPlatformPkg/Bds/Bds.c
@@ -16,6 +16,7 @@
 
 #include <Library/PcdLib.h>
 #include <Library/PerformanceLib.h>
+#include <Library/DevicePathLib.h>
 
 #include <Protocol/Bds.h>
 #include <Protocol/SimpleFileSystem.h>
@@ -230,6 +231,7 @@ FindCandidateOnHandle (
     L"\\EFI\\fedora\\grubaa64.efi",
     NULL
   };
+  CHAR16                          *DevicePathString;
 
   Status = gBS->HandleProtocol (Handle, &gEfiSimpleFileSystemProtocolGuid,
                   (VOID **) &FileSystem);
@@ -262,7 +264,13 @@ FindCandidateOnHandle (
     goto CloseRoot;
   }
 
-  DEBUG ((EFI_D_INFO, "%a: found \"%s\"\n", __FUNCTION__, *FileName));
+  DevicePathString = ConvertDevicePathToText (*Candidate,
+                       FALSE /* DisplayOnly */, FALSE /* AllowShortcuts */);
+  DEBUG ((EFI_D_INFO, "%a: found \"%s\"\n", __FUNCTION__,
+    DevicePathString == NULL ? *FileName : DevicePathString));
+  if (DevicePathString != NULL) {
+    FreePool (DevicePathString);
+  }
 
 CloseRoot:
   RootDir->Close (RootDir);
diff --git a/ArmPlatformPkg/Bds/Bds.inf b/ArmPlatformPkg/Bds/Bds.inf
index bd69259..4d37e2a 100644
--- a/ArmPlatformPkg/Bds/Bds.inf
+++ b/ArmPlatformPkg/Bds/Bds.inf
@@ -48,6 +48,7 @@
   PrintLib
   BaseLib
   NetLib
+  DevicePathLib
 
 [Guids]
   gEfiEndOfDxeEventGroupGuid
-- 
1.8.3.1

